core/tests/vm-benchmark/benches/criterion.rs

```diff
@@ -1,20 +1,99 @@
-use criterion::{black_box, criterion_group, criterion_main, Criterion};
-use zksync_vm_benchmark_harness::{cut_to_allowed_bytecode_size, get_deploy_tx, BenchmarkingVm};
+use std::time::Duration;
+
+use criterion::{
+    black_box, criterion_group, criterion_main, measurement::WallTime, BatchSize, BenchmarkGroup,
+    Criterion,
+};
+use zksync_types::Transaction;
+use zksync_vm_benchmark_harness::{
+    cut_to_allowed_bytecode_size, get_deploy_tx, get_heavy_load_test_tx, get_load_test_deploy_tx,
+    get_load_test_tx, get_realistic_load_test_tx, BenchmarkingVm, BenchmarkingVmFactory, Fast,
+    Legacy, LoadTestParams,
+};
+
+const SAMPLE_SIZE: usize = 20;
+
+fn benches_in_folder<VM: BenchmarkingVmFactory, const FULL: bool>(c: &mut Criterion) {
+    let mut group = c.benchmark_group(VM::LABEL.as_str());
+    group
+        .sample_size(SAMPLE_SIZE)
+        .measurement_time(Duration::from_secs(10));
 
-fn benches_in_folder(c: &mut Criterion) {
     for path in std::fs::read_dir("deployment_benchmarks").unwrap() {
         let path = path.unwrap().path();
 
         let test_contract = std::fs::read(&path).expect("failed to read file");
 
         let code = cut_to_allowed_bytecode_size(&test_contract).unwrap();
         let tx = get_deploy_tx(code);
-
-        c.bench_function(path.file_name().unwrap().to_str().unwrap(), |b| {
-            b.iter(|| BenchmarkingVm::new().run_transaction(black_box(&tx)))
+        let file_name = path.file_name().unwrap().to_str().unwrap();
+        let full_suffix = if FULL { "/full" } else { "" };
+        let bench_name = format!("{file_name}{full_suffix}");
+        group.bench_function(bench_name, |bencher| {
+            if FULL {
+                // Include VM initialization / drop into the measured time
+                bencher.iter(|| BenchmarkingVm::<VM>::default().run_transaction(black_box(&tx)));
+            } else {
+                bencher.iter_batched(
+                    BenchmarkingVm::<VM>::default,
+                    |mut vm| {
+                        let result = vm.run_transaction(black_box(&tx));
+                        // FIXME: transactions mostly run out of gas here; is this OK?
```

Comment:
Subj. All transactions other than `deploy_simple_contract` fail with an out-of-gas error. Is this expected / OK?

Reply:
Yes. This way they can be tested with different gas limits to figure out if the time increases roughly linearly as it should.
Also, some of the most degenerate edge cases cannot be stopped any other way.